blogs/2025/07/15/copilot-coding-agent.md

```diff
@@ -0,0 +1,88 @@
+
+The agent runs within a temporary, isolated dev environment (similar to GitHub Actions) that gets spun up where the agent can explore the codebase, make changes, build the code, run tests, etc. - a complete dev enviornment just for the agent so it can function in a fully autonomous manner.
+
+And now, we're integrating the coding agent and all of its workflows directly into VS Code.
+
+## Copilot Coding Agent in Visual Studio Code
+
+Copilot Coding Agent integrates with the [GitHub Pull Requests extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github). You can assign any issue or PR to Copilot Coding Agent right from the sidebar - no need to switch to your browser. Just select, assign, and let the agent take it from there.
+
+![A GIF showing Copilot assigned to an issue from within VS Code](assign-to-copilot-gif.gif)
+
+### Track your agent’s progress
+
+There’s a new **Copilot on My Behalf** query in the Pull Requests view. This shows you everything Copilot is working on for you. Want to see exactly what the agent did? Click "View Session" and watch the play-by-play. You get full transparency into every command and decision. You can also terminate the agent if you're not happy with how things are going. And you can drive all of this without even leaving the editor.
+
+![A WIP PR shown under the "Copilot on my behalf" section with the "View Session" button highlighted](copilot-wip.png)
+
+In this screenshot, the agent implemented a "trending" section to a website, complete with UI and database changes. All of this is one interaction so it's only one premium request.
+
+### Review, comment, and iterate
+
+When the agent is done, it will assign the PR to you and request you as a reviewer. It even includes a screenshot if applicable so you can quickly verify whether it did the right thing or not at all.
+
+![PR from Coding agent shown in VS Code with a screenshot](draft-with-screenshot.png)
+
+You can review the agent’s PR, leave comments, and ask for changes - all from within VS Code. The agent will pick up your feedback, update the PR, and let you know when it's finished.
+
+This workflow integrates beautifully with services that offer previews for pull requests.
+For example, if you use Azure Static Web Apps, Vercel, Netlify, and others, you can review the agent's work without even having to check out and run a branch locally.
+
+![Live preview of the site on Netlify but we have an error](live-preview-netlify.png)
+
+Uh oh - now that we look closely, it looks like we didn't quite get the "Trending" section working. In this case, that's ok. In fact it's good. Our agent couldn't succeed in one shot because it doesn't have access to create the required database changes. Instead, so it created a migration script as part of the PR. That *will* require us to manually check out the PR, but you might feel more comfortable being in control over that sort of change.
```

Comment:
```suggestion
Uh oh - now that we look closely, it looks like we didn't quite get the "Trending" section working. In this case, that's ok. In fact it's good. Our agent couldn't succeed in one shot because it doesn't have access to create the required database changes. Instead, it created a migration script as part of the PR. That *will* require us to manually check out the PR, but you might feel more comfortable being in control over that sort of change.
```